src/lfview/lf-data-provider.ts

```diff
@@ -0,0 +1,487 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { LanguageClient } from 'vscode-languageclient';
+
+/**
+ * Defines the different roles of nodes that can be displayed in the LFDataProvider tree view.
+ * {@code ROOT}: Represents the root node of the tree view.
+ * {@code FILE}: Represents a file node in the tree view.
+ * {@code REACTOR}: Represents a reactor node in the tree view.
+ */
+export enum LFDataProviderNodeRole {
+    ROOT = 'root',
+    FILE = 'file',
+    REACTOR = 'reactor'
+}
+
+/**
+ * Defines the types of the displayed data provider tree view.
+ * 
+ * {@code LOCAL} represents nodes for local LF libraries.
+ * {@code LIBRARY} represents nodes for LF libraries downloaded using Lingo.
+ */
+export enum LFDataProviderNodeType {
+    LOCAL = 1,
+    LIBRARY = 2
+}
+
+/**
+ * Represents a node in the LFDataProvider tree.
+ * 
+ * @param label - The label to display for the node.
+ * @param uri - The URI of the resource associated with the node.
+ * @param role - The role of the node, such as LFDataProviderNodeType.REACTOR, LFDataProviderNodeType.FILE, or LFDataProviderNodeType.ROOT.
+ * @param children - The child nodes of this node, if any.
+ * @param position - The start and end positions of the node in the source code, if available.
+ */
+export class LFDataProviderNode extends vscode.TreeItem {
+
+    children: LFDataProviderNode[] | undefined;
+    role: string;
+    position: NodePosition | undefined;
+    type: LFDataProviderNodeType;
+
+    /**
+     * Represents the URI of the data provider node.
+     * Avoid using TreeItem.resourceUri to maintain a clear separation between error detection and Tree Item rendering.
+     */
+    uri: vscode.Uri;
+
+    constructor(label: string, uri: string, role: string, type: LFDataProviderNodeType,
+        children?: LFDataProviderNode[] | undefined, position?: NodePosition | undefined) {
+        let newLabel = label.replace('.lf', '');
+        super(newLabel, role === LFDataProviderNodeRole.REACTOR ? vscode.TreeItemCollapsibleState.None : vscode.TreeItemCollapsibleState.Collapsed);
+        this.uri = vscode.Uri.parse(uri);
+        this.children = children;
+        this.role = role;
+        this.type = type;
+        let icon = role === LFDataProviderNodeRole.ROOT ? 'root-folder' : role === LFDataProviderNodeRole.FILE ? 'file-code' : 'json';
+        this.iconPath = new vscode.ThemeIcon(icon);
+        this.contextValue = role;
+        if (position) { this.position = position; }
+
+        // this.registerNodeCommand();
```

Comment:
Maybe remove this, since it is commented out or add a FIXME explaining what to do here instead.

Reply:
Initially, I added a command that would open the specific file URI when clicking the Reactor Tree item. However, I found that it was quite annoying because it would open the file on the side every time you selected the item. Instead, I added an action, which is a much better approach. I’ll remove the previous code.